### ISSUE TITLE
When high number of albums browser hangs due to expanded albums.

### ISSUE THREAD

**Issue by ronbruins** [immich-app/immich#11071]
### The bug

I created a large number of albums, which are grouped by year. When going on te web to the album page, by default the album list is expanded. This causes the browser to apparently load all thumbnails and as such gets stuck and so the search function doesn't work.

When collapsing all albums and/or put it in list mode, the strain on the browser is less, therefore it could be an idea to by default show the list view, or have the groups collapse. 

### The OS that Immich Server is running on

Docker

### Version of Immich Server

v108

### Version of Immich Mobile App

latest

### Platform with the issue

- [ ] Server
- [X] Web
- [x] Mobile

### Your docker-compose.yml content

```YAML
NA
```

### Your .env content

```Shell
NA
```

### Reproduction steps

```bash
1.create loads (shared) albums and group them.
2. go to the album page 
3. try to search and it hangs

workaround:
2. go to the album page 
3. collapse all or change to list
```

**Comment by alextran1502** [comment]
How many albums do you have to started noticing the issue?

**Comment by ronbruins** [comment]
> How many albums do you have to started noticing the issue?

A lot. Maybe hundreds, I deleted them all and started again per year. I renamed my folders with date and meaning full words / description and created a python script to automatically create them based on the folder name. So I can search for events, which is otherwise not possible. 

It works well in list or collapsed mode.

**Comment by alextran1502** [comment]
Thank, we will reproduce and optimize the page

**Comment by ronbruins** [comment]
> Thank, we will reproduce and optimize the page
> 
> 

Currently I am on 90 albums and web seems to be still fine. Will test it a bit more. 

However mobile is struggling already and lost its snappy feeling until (it seems) its local database is synced or so. 
Scrolling becomes less smooth and after a while its fine.

**Comment by ronbruins** [comment]
I have recreated a bunch of albums, after first assessing my library. Using my python script, all folders ending with # go to an automated created 'YYYY misc photos' Where others ending with @ get their own album. 

I am now at 161 albums and the performance is still OK. I have covered 2024-2020 (before I had it all the way till 2005) so long way to go.

**Comment by heymrdjcw** [comment]
I appreciate that you opened this because I was coming to do the same. I've had the same issue since 102.1 (which is the first version I started using). I finally noticed that if I set the album page to list, I can then use search and not hang the browser. Otherwise it hangs the browser. 

I have 531,000 images and 11,700 videos spread over 12,000 albums.

**Comment by ronbruins** [comment]
I am now up to 546 albums and I start to see the slowness when all albums are expanded.

**Comment by dlang123** [comment]
I have about 7000 albums and it is almost unusable in how slow it is.  When I initially added all of my photos I let it create albums based on folder names because some of the folder names contained dates that needed to be corrected in exif data and some folders were actual albums from lightroom.  I am currently working to trim my albums and folders down but it will take a while to collapse them all into some that are usable. I currently have 196850 pictures and 6826 videos and 3.43TB of storage used for Immich and I haven't even started adding the wife's pictures which will probably almost double the number of photos.

**Comment by danieldietzler** [comment]
Hey, the album queries have been optimized recently. Is this better now?

**Comment by dlang123** [comment]
> Hey, the album queries have been optimized recently. Is this better now?

Yes it is world's better.  I am not sure how many albums I have but the recent optimizations has made the album page useful again.

I don't know how many albums I have now, but I do have over 300k pictures, 12k videos (some 8k), 4TB used and the whole thing is pretty responsive.

**Comment by danieldietzler** [comment]
Awesome! Closing this then :)
